### PR TITLE
Feature/720 archive dag instances

### LINKED ISSUE #720: Create sql script to archive dag instances
In order to keep the dag_instances, job_instances and event tables small, dag instances (and the corresponding job instances and events) older than e.g. 6 months should be moved to separate archive tables. The job parameters column of job instance should not be moved, but deleted instead, to save space in the DB.

### PR DESCRIPTION
**Pull request by kevinwallimann**
Closes #720 

The procedure can be called e.g. like this:

```
CALL archive_dag_instances(
    i_to_ts => (now() - interval '12 months')::timestamp,
    i_max_records => 200000,
    i_chunk_size => 40000
);
```

### CODE REVIEW

File: src/main/resources/db_scripts/liquibase/v0.5.13.add-archive-procedure.sql [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    DROP TABLE dag_instance_ids_to_archive;
+
+    RAISE NOTICE '=============';
+    RAISE NOTICE '  END BATCH';
+    RAISE NOTICE '=============';
+END;
+$$ LANGUAGE plpgsql;
+
+
+CREATE INDEX IF NOT EXISTS event_dag_instance_idx ON "event" (dag_instance_id);
```
Comment (jozefbakus) [inline src/main/resources/db_scripts/liquibase/v0.5.13.add-archive-procedure.sql:156]: Missing in db_script_latest

File: src/main/resources/db_scripts/liquibase/v0.5.13.add-archive-procedure.sql [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE PROCEDURE archive_dag_instances(
+    IN i_from_ts timestamp default to_timestamp('1970-01-01', 'yyyy-mm-dd'),
+    IN i_to_ts timestamp default now() - interval '6 months',
+    IN i_max_records int default 50000000,
```
Comment (jozefbakus) [inline src/main/resources/db_scripts/liquibase/v0.5.13.add-archive-procedure.sql:19]: Different defaults in latest db script

**Review by jozefbakus — DISMISSED** [review]
Other than two comments looks good

File: src/main/resources/db_scripts/liquibase/v0.5.13.add-archive-procedure.sql (hunk 2 above)
Comment (kevinwallimann) [reply src/main/resources/db_scripts/liquibase/v0.5.13.add-archive-procedure.sql:19]: Good catch!

File: src/main/resources/db_scripts/liquibase/v0.5.13.add-archive-procedure.sql (hunk 1 above)
Comment (kevinwallimann) [reply src/main/resources/db_scripts/liquibase/v0.5.13.add-archive-procedure.sql:156]: Here it is
image

I added it before the procedures to keep all the indexes together. But it seems to me that we're missing some indexes anyway.

**Review by jozefbakus — DISMISSED** [review]

**Review by jozefbakus — APPROVED** [review]

**Pull request merged**